src/mbgl/gl/object_store.hpp

```diff
@@ -75,8 +66,13 @@ class ObjectStore : private util::noncopyable {
     }
 
     UniqueTexture createTexture() {
-        GLuint id = 0;
-        MBGL_CHECK_ERROR(glGenTextures(1, &id));
+        if (pooledTextures.empty()) {
+            pooledTextures.resize(TextureMax);
+            MBGL_CHECK_ERROR(glGenTextures(TextureMax, pooledTextures.data()));
+        }
+
+        GLuint id = pooledTextures.back();
```

Comment:
I think this this will fail once the amount of textures in use reaches `TextureMax + 1`.

Previous texture pool logic uses a _pool of pools_, where a new texture pool would be generated when the current pool size reaches its limit - and old & unused pools would be eventually free'd.